nixos/modules/programs/dconf.nix

```diff
@@ -4,53 +4,55 @@ with lib;
 
 let
   cfg = config.programs.dconf;
-  cfgDir = pkgs.symlinkJoin {
-    name = "dconf-system-config";
-    paths = map (x: "${x}/etc/dconf") cfg.packages;
-    postBuild = ''
-      mkdir -p $out/profile
-      mkdir -p $out/db
-    '' + (
-      concatStringsSep "\n" (
-        mapAttrsToList (
-          name: path: ''
-            ln -s ${path} $out/profile/${name}
-          ''
-        ) cfg.profiles
-      )
-    ) + ''
-      ${pkgs.dconf}/bin/dconf update $out/db
-    '';
-  };
+
+  asFileDb = val:
+    let db = with builtins;
```

Comment:
We are slowly phasing out `with` statement in GNOME module and package expressions, since it makes it harder to reason about where do attributes come from.

Please do not introduce new ones and instead use full attribute path where not too noisy, or appropriately scoped `let inherit` otherwise:

https://github.com/NixOS/nixpkgs/blob/52f9de5b39a22c86899d6e35c68ac7fab42dbb98/nixos/modules/services/desktops/gnome/gnome-browser-connector.nix#L4